onnxruntime/core/optimizer/optimizer_execution_frame.cc

```diff
@@ -90,7 +90,7 @@ AllocatorPtr OptimizerExecutionFrame::GetAllocatorImpl(const OrtAllocatorInfo& i
 // This method is not thread safe!
 // Return S_OK and nullptr if index map to an value that is an unused optional input/output
 Status OptimizerExecutionFrame::CreateNodeOutputMLValueImpl(OrtValue& ort_value, int ort_value_idx,
-                                                            const TensorShape* shape) {
+                                                            const TensorShape* shape, size_t nnz) {
   const DataTypeImpl* ml_type = utils::GetMLDataType(*(info_.GetMLValueIdxNodeArgMap().at(ort_value_idx)));
```

Comment:
should nnz be marked as unused here with ORT_UNUSED_PARAMETER?

Reply:
Added the sparse-tensor allocation logic to the optimizer execution frame too.